### ISSUE TITLE
Function "Create table to keep arguments for additional 'menu' info" not work in javascriptcomplete

### ISSUE THREAD

**Issue by ava57r** [vim/vim#1462]
Good day.
I found that code in the string
https://github.com/vim/vim/blob/master/runtime/autoload/javascriptcomplete.vim#L566
does not work.
function matchlist() return a list with 10 elements always and does not depend on {pat}
Vim release - 7.4.1689

**Comment by chrisbra** [comment]
Yeah, that looks certainly wrong. Please try to contact the maintainer of the file. However I haven't seen Mikolaj in a long time, so it might be that he is not available anymore.

**Comment by ava57r** [comment]
I tested a fix
you must replace `if len(f_elements) == 3` `if len(f_elements) > 0` will work.

**Comment by chrisbra** [comment]
did you try to contact the maintainer?

**Comment by ava57r** [comment]
I wrote an e-mail. Waiting for an answer.

**Comment by ava57r** [comment]
Hi. Response from Mikolaj Machowski no.

**Comment by brammool** [comment]
Is this about this code:

	let f_elements = matchlist(i, 'function\s\+\(\k\+\)\s*(\(.\{-}\))')
		if len(f_elements) == 3
			let b:js_menuinfo[f_elements[1].'('] = f_elements[2]
		endif

It is using the third element, thus it should probably be:
		if len(f_elements) >= 3

**Comment by ava57r** [comment]
Test run this code:
```vim
let f_elements = matchlist('function CreateIB(sPath, sFile, sIBDir, sOutLog)', 'function\s\+\(\k\+\)\s*(\(.\{-}\))')
```
Print value f_elements:
```vim
['function CreateIB(sPath, sFile, sIBDir, sOutLog)', 'CreateIB', 'sPath, sFile, sIBDir, sOutLog', '', '', '', '', '',
 '', '']
``` 
When there is no match an empty list is returned.

this code work:
```vim
if len(f_elements) > 0
```
 this code work:
```vim
if len(f_elements) >= 3
```

**Comment by chrisbra** [comment]
I think this has been fixed with 214641f77df6f318a
